### ISSUE TITLE
_set_attrs_to_values(response) does not update attributes anymore?

### ISSUE THREAD

**Issue by prithugoswami** [celiao/tmdbsimple#45]
I am new to python and am really sorry if this is a very trivial thing or I have got it wrong in some way.  
When we make a search request using:  
```python
search = tmdb.Search()  
response = search.movie(query='Harry Potter')
```
we get to use `search.results` to access the results.  
But after I call the `Search.movie()` method again, the attributes do not get updated.  
  
Example:  
I first make a query of 'The Matrix' and then a query for 'Star Wars' and then pretty-print the first item in `search.results` and `response['results']` respectively.
```python
>>> import pprint as p
>>> import tmdbsimple as tmdb
>>> tmdb.API_KEY = 'xyz'
>>> search = tmdb.Search()
>>> response = search.movie(query='Matrix')
>>> p.pprint(search.results[0])
{'adult': False,
 'backdrop_path': '/7u3pxc0K1wx32IleAkLv78MKgrw.jpg',
 'genre_ids': [28, 878],
 'id': 603,
 'original_language': 'en',
 'original_title': 'The Matrix',
 'overview': 'Set in the 22nd century, The Matrix tells the story of a '
             'computer hacker who joins a group of underground insurgents '
             'fighting the vast and powerful computers who now rule the earth.',
 'popularity': 31.32734,
 'poster_path': '/hEpWvX6Bp79eLxY1kX5ZZJcme5U.jpg',
 'release_date': '1999-03-30',
 'title': 'The Matrix',
 'video': False,
 'vote_average': 7.9,
 'vote_count': 9830}
>>> p.pprint(response['results'][0])
{'adult': False,
 'backdrop_path': '/7u3pxc0K1wx32IleAkLv78MKgrw.jpg',
 'genre_ids': [28, 878],
 'id': 603,
 'original_language': 'en',
 'original_title': 'The Matrix',
 'overview': 'Set in the 22nd century, The Matrix tells the story of a '
             'computer hacker who joins a group of underground insurgents '
             'fighting the vast and powerful computers who now rule the earth.',
 'popularity': 31.32734,
 'poster_path': '/hEpWvX6Bp79eLxY1kX5ZZJcme5U.jpg',
 'release_date': '1999-03-30',
 'title': 'The Matrix',
 'video': False,
 'vote_average': 7.9,
 'vote_count': 9830}
>>> response = search.movie(query='Star Wars')
>>> p.pprint(search.results[0])
{'adult': False,
 'backdrop_path': '/7u3pxc0K1wx32IleAkLv78MKgrw.jpg',
 'genre_ids': [28, 878],
 'id': 603,
 'original_language': 'en',
 'original_title': 'The Matrix',
 'overview': 'Set in the 22nd century, The Matrix tells the story of a '
             'computer hacker who joins a group of underground insurgents '
             'fighting the vast and powerful computers who now rule the earth.',
 'popularity': 31.32734,
 'poster_path': '/hEpWvX6Bp79eLxY1kX5ZZJcme5U.jpg',
 'release_date': '1999-03-30',
 'title': 'The Matrix',
 'video': False,
 'vote_average': 7.9,
 'vote_count': 9830}
>>> p.pprint(response['results'][0])
{'adult': False,
 'backdrop_path': '/5Iw7zQTHVRBOYpA0V6z0yypOPZh.jpg',
 'genre_ids': [28, 12, 14, 878],
 'id': 181808,
 'original_language': 'en',
 'original_title': 'Star Wars: The Last Jedi',
 'overview': 'Rey develops her newly discovered abilities with the guidance of '
             'Luke Skywalker, who is unsettled by the strength of her powers. '
             'Meanwhile, the Resistance prepares to do battle with the First '
             'Order.',
 'popularity': 778.931034,
 'poster_path': '/xGWVjewoXnJhvxKW619cMzppJDQ.jpg',
 'release_date': '2017-12-13',
 'title': 'Star Wars: The Last Jedi',
 'video': False,
 'vote_average': 7.3,
 'vote_count': 2167}

```

**Comment by prithugoswami** [comment]
@3lixy But can't we get rid of the check `if not hasattr(self, key)` completely?  
That would set the attributes every time.

**Comment by prithugoswami** [comment]
I just had a look at Issue #30 and realised commit https://github.com/celiao/tmdbsimple/commit/1d99a1875110a9c8d1e4162faf65c53b6ab8c923 adds the `if not hasattr(self, key)` check. I do not completely understand what it fixes.  
But adding to @3lixy suggestion, adding all the keys that are returned in a search query would do the trick.  

`if not hasattr(self, key) or key in {'results', 'page', 'total_results', 'total_pages'}:`  
I think having this for the Search class is necessary as most of the time multiple search queries are made one after the other.

**Comment by 3lixy** [comment]
Yea i don;t know why if not hasattr(self, key) is there in the first place. I think we need @celiao to comment on that.

**Comment by celiao** [comment]
Nice find!

`if not hasattr(self, key):` was added for the case where a method returned a dict with a key of the same name as the method.  For example, `Movie.keywords()` returns a dict with a key called `keyword`.  If the attribute was set, then subsequent calls to `Movie.keywords()` would raise an error.

The problem with `if not hasattr(self, key):`, as you discovered, is that subsequent method calls don't reset the attributes, as they are already set.

To fix the problem for both issues, I added a check to see if the "attribute" is "callable()", i.e., 
`if not hasattr(self, key) or not callable(getattr(self, key)):`.

To confirm all is well, for now and in the future, I created a `tests/test_base.py` file.  I moved the multiple calls to `Movies.keywords()` test into that file and added another test for your `Search.movie()` example.
